Scheduled teardown worker secrets: errors on re-run

## Bug: Worker secret setup fails with "Binding name already in use" on re-run

### Description

The "Set Scheduled Teardown Worker secrets" step fails for several bindings when
`setup-control-plane` is re-run on an existing deployment:

```
✘ [ERROR] Binding name 'GITHUB_OWNER' already in use.
Please use a different name and try again. [code: 10053]
```


The same error occurs for `GITHUB_REPO`, `DOMAIN`, and `ADMIN_EMAIL`.
`GITHUB_TOKEN` and `RESEND_API_KEY` succeed.

### Root Cause

Cloudflare error 10053 means the binding name is already in use by a **different
binding type** — not a duplicate secret. The failing names (`GITHUB_OWNER`,
`GITHUB_REPO`, `DOMAIN`, `ADMIN_EMAIL`) are likely defined as plain environment
variable bindings in the Worker's Terraform configuration. Cloudflare does not allow
a secret to share a name with an existing env var binding.

`GITHUB_TOKEN` and `RESEND_API_KEY` succeed because they are not defined as env var
bindings in Terraform — only as secrets.

### Impact

The worker runs with stale values for the affected bindings. On first-time setup
this is not a problem, but re-running `setup-control-plane` (e.g. to rotate
credentials or fix a failed step) leaves the worker in an inconsistent state
without any clear error surfaced to the user — the step ends with
`✅ Worker secrets configured` despite the failures.

### Expected Behavior

Each value should be configured in exactly one place — either as a Terraform env var
binding or as a Wrangler secret, not both. The step should also fail loudly if any
secret cannot be set, rather than printing success after `|| echo "⚠️  Failed"`.



## Errors observed in the log:

Run echo "⚙️  Setting Scheduled Teardown Worker secrets..."
⚙️  Setting Scheduled Teardown Worker secrets...
  Setting GITHUB_TOKEN...
npm warn exec The following package was not found and will be installed: wrangler@4.78.0

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 
✨ Success! Uploaded secret GITHUB_TOKEN
  Setting GITHUB_OWNER...

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 

✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/nexus-odeslab-com-worker/secrets) failed.

  Binding name 'GITHUB_OWNER' already in use. Please use a different name and try again. [code: 10053]
  
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose


🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-03-29_13-17-47_557.log"
⚠️  Failed
  Setting GITHUB_REPO...

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 

✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/nexus-odeslab-com-worker/secrets) failed.

  Binding name 'GITHUB_REPO' already in use. Please use a different name and try again. [code: 10053]
  
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose


🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-03-29_13-17-50_595.log"
⚠️  Failed
  Setting DOMAIN...

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 

✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/nexus-odeslab-com-worker/secrets) failed.

  Binding name 'DOMAIN' already in use. Please use a different name and try again. [code: 10053]
  
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose


🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-03-29_13-17-53_079.log"
⚠️  Failed
  Setting RESEND_API_KEY...

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 
✨ Success! Uploaded secret RESEND_API_KEY
  Setting ADMIN_EMAIL...

 ⛅️ wrangler 4.78.0
───────────────────
🌀 Creating the secret for the Worker "nexus-odeslab-com-worker" 

✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/nexus-odeslab-com-worker/secrets) failed.

  Binding name 'ADMIN_EMAIL' already in use. Please use a different name and try again. [code: 10053]
  
  If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose


🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-03-29_13-17-58_408.log"
⚠️  Failed
✅ Worker secrets configured

--- a/.github/workflows/setup-control-plane.yaml
+++ b/.github/workflows/setup-control-plane.yaml
@@ -231,14 +231,13 @@ jobs:
               else
                 echo "⚠️  Failed to auto-save secrets: $SECRET_ERROR"
                 echo ""
-                echo "╔══════════════════════════════════════════════════════════════════════════════╗"
-                echo "║  Save these credentials manually as GitHub Secrets:                        ║"
-                echo "╠══════════════════════════════════════════════════════════════════════════════╣"
-                echo "║  R2_ACCESS_KEY_ID=$R2_ACCESS_KEY_ID"
-                echo "║  R2_SECRET_ACCESS_KEY=$R2_SECRET_ACCESS_KEY"
-                echo "╚══════════════════════════════════════════════════════════════════════════════╝"
+                echo "  Please save these credentials manually as GitHub Secrets:"
                 echo ""
-                echo "Or configure GH_SECRETS_TOKEN with 'repo' scope and 'Secrets: Read and Write' permission"
+                echo "  R2_ACCESS_KEY_ID=$R2_ACCESS_KEY_ID"
+                echo "  R2_SECRET_ACCESS_KEY=$R2_SECRET_ACCESS_KEY"
+                echo ""
+                echo "  To enable auto-saving, configure GH_SECRETS_TOKEN:"
+                echo "  Fine-grained PAT with Secrets (Read and Write) permission"
               fi
             else
               echo ""

--- a/docs/setup-guide.md
+++ b/docs/setup-guide.md
@@ -124,7 +124,7 @@ Add these secrets to your GitHub repository:
 
 | Secret Name | Description |
 |-------------|-------------|
-| `GH_SECRETS_TOKEN` | GitHub PAT for auto-saving R2 credentials (see below) |
+| `GH_SECRETS_TOKEN` | GitHub PAT for R2 auto-save and Cloudflare runtime (see below) |
 | `TF_VAR_user_email` | User - all services except SSH |
 | `RESEND_API_KEY` | Email notifications via Resend |
 | `DOCKERHUB_USERNAME` | Docker Hub username (higher pull limits) |
@@ -132,13 +132,15 @@ Add these secrets to your GitHub repository:
 
 #### GH_SECRETS_TOKEN
 
-This token allows the initial setup workflow to automatically save R2 credentials as GitHub Secrets. Without it, you must manually copy the credentials from the workflow logs after the first run.
+This token allows the initial setup workflow to automatically save R2 credentials as GitHub Secrets. It is also used as the runtime `GITHUB_TOKEN` in Cloudflare (for the scheduled teardown worker and Control Plane), so it must be able to dispatch workflows. Without it, you must manually copy the credentials from the workflow logs after the first run, and Cloudflare-based automation that triggers GitHub Actions will fail.
 
 **How to create:**
 1. Go to **GitHub** → **Settings** → **Developer settings** → **Personal access tokens** → **Fine-grained tokens**
 2. Click **"Generate new token"**
 3. **Repository access**: Select your Nexus-Stack repository
-4. **Permissions**: Repository permissions → **Secrets** → **Read and Write**
+4. **Permissions** (Repository permissions):
+   - **Secrets** → **Read and write**
+   - **Actions** → **Read and write** (required so Cloudflare workers can dispatch workflows)
 5. Copy the token and save it as `GH_SECRETS_TOKEN` in your repository secrets
 
 ### Optional Repository Variables

